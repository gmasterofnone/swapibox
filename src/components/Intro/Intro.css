
.fade {
  position: relative;
  width: 100%;
  min-height: 40vh;
  top: -120px;
  background-image: linear-gradient(0deg, transparent, black 75%);
  z-index: 1;
}

.star-wars {
  display: flex;
  justify-content: center;
  position: relative;
  height: 350px;
  color: #feda4a;
  font-family: 'Pathway Gothic One', sans-serif;
  font-size: 600%;
  font-weight: 600;
  letter-spacing: 6px;
  line-height: 150%;
  perspective: 325px;
  text-align: justify;
}

.crawl {
  position: relative;
  top: 99999px;
  transform-origin: 40% 70%;
  animation: crawl 40s linear;
}

.crawl > .title {
  font-size: 125%;
  text-align: center;
}

.crawl > .title h1 {
  margin: 0 0 150px;
  line-height: 100%;
  text-transform: uppercase;
}

@keyframes crawl {
    0% {
    top: -100px;
    transform: rotateX(20deg)  translateZ(0);
  }
    100% { 
    top: -6000px;
    transform: rotateX(25deg) translateZ(-2500px);
  }
}