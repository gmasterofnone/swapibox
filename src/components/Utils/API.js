

export const getIntro = async () => {
  const url = 'films/'
  const result = await fetchData(url)  
  return result;
}


export const fetchData = async (url, func = null) => {
  const response = await fetch(`https://swapi.co/api/${url}`);
  const intro = await response.json();
  return intro.results;  
}




