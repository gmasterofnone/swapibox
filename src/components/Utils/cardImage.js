const cardImage = {
  'Luke Skywalker': require('../Utils/assets/cards/luke-skywalker.jpg'),
  'C-3PO': require('../Utils/assets/cards/c-3po.jpg'),
  'R2-D2': require('../Utils/assets/cards/r2-d2.jpg'),
  'Darth Vader': require('../Utils/assets/cards/darthvader.jpg'),
  'Leia Organa': require('../Utils/assets/cards/Leia.jpg'),
  'Owen Lars': require('../Utils/assets/cards/uncle.jpg'),
  'Beru Whitesun lars': require('../Utils/assets/cards/beru.jpg'),
  'R5-D4': require('../Utils/assets/cards/r5-d4.jpg'),
  'Biggs Darklighter': require('../Utils/assets/cards/biggs.jpg'),
  'Obi-Wan Kenobi': require('../Utils/assets/cards/obi-wan.jpg'),
  'Alderaan': require('../Utils/assets/cards/alderaan.jpg'),
  'Yavin IV': require('../Utils/assets/cards/yavin.jpg'),
  'Hoth': require('../Utils/assets/cards/hoth.jpg'),
  'Dagobah': require('../Utils/assets/cards/dagobah.jpg'),
  'Bespin': require('../Utils/assets/cards/bespin.jpg'),
  'Endor': require('../Utils/assets/cards/endor.jpg'),
  'Naboo': require('../Utils/assets/cards/noboo.jpg'),
  'Coruscant': require('../Utils/assets/cards/cruscant.jpg'),
  'Kamino': require('../Utils/assets/cards/kamino.jpg'),
  'Geonosis': require('../Utils/assets/cards/geonosis.jpg'),
  'Sand Crawler': require('../Utils/assets/cards/sand.jpg'),
  'T-16 skyhopper': require('../Utils/assets/cards/skyhopper.jpg'),
  'X-34 landspeeder': require('../Utils/assets/cards/landspeeder.jpg'),
  'TIE/LN starfighter': require('../Utils/assets/cards/tie.jpg'),
  'Snowspeeder': require('../Utils/assets/cards/snowspeed.jpg'),
  'TIE bomber': require('../Utils/assets/cards/bomber.jpg'),
  'AT-AT': require('../Utils/assets/cards/at-at.jpg'),
  'AT-ST': require('../Utils/assets/cards/at-st.jpg'),
  'Storm IV Twin-Pod cloud car': require('../Utils/assets/cards/pod.jpg'),
  'Sail barge': require('../Utils/assets/cards/sail.jpg'),
  true: require('../Utils/assets/cards/star-true.svg'),
  false: require('../Utils/assets/cards/star-false.svg')
}

export default cardImage;