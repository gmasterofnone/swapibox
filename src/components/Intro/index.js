import React from 'react';
import 'react-star-wars-crawl/lib/index.css'
import './Intro.css';
import PropTypes from 'prop-types';

const Intro = ({ films }) => {
  const randomNumber = Math.round(Math.random() * films.length);
  
  return (
    <div>
    <div className="fade"></div>
      <section className="star-wars">
        <div className="crawl">
          <div className="title">
            <h1>{films[randomNumber].title}</h1>
            <p>{films[randomNumber].release_date}</p> 
          </div>
          <p>{films[randomNumber].opening_crawl}</p>  
        </div>
      </section>
    </div>
  )
}

Intro.propTypes = {
  films: PropTypes.array.isRequired
}

export default Intro;