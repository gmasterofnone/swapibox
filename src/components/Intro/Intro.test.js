import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Intro from '.';
import {films} from '../Utils/mockData'

describe('Intro', () => {
  let wrapper;
  let intro = [films.results[0]]

  beforeEach(() => {
    wrapper = shallow(<Intro intro={intro} />);
  });


  it('matches the snapshot', () => { 

    expect(wrapper).toMatchSnapshot()
  });

});