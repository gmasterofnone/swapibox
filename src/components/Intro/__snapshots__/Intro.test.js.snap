// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intro matches the snapshot 1`] = `undefined`;
